### ISSUE TITLE
Allow input of arbitrary shape

### ISSUE THREAD

**Issue by sndre** [tf-coreml/tf-coreml#154]
I'm trying to convert fully-convolutional network from TensorFlow to CoreML. In TensorFlow, the input can be an image of arbitrary size hence the initialization for Placeholder is as follows:

```python
xs = tf.placeholder(tf.float32, [1, None, None, 3])
... other layers
```

When trying to convert this network to CoreML:
```python
tf_converter.convert(tf_model_path = 'graph.pb',
                     mlmodel_path = 'model.mlmodel',
                     output_feature_names = ['BiasAdd:0'],
                     image_input_names = ['Placeholder:0'])
```

I get the following error:
```
AssertionError: Placeholder:0 is a placeholder with incomplete shape <unknown>.
```

Does it mean that CoreML doesn't support fully-convolutional networks with input of arbitrary size?

**Comment by aseemw** [comment]
No, it doesn't. 
At the moment, CoreML requires a fixed input size to be baked in.

**Comment by sndre** [comment]
@aseemw, thanks for the quick reply! 🙌 Any tentative plans to support it?

**Comment by aseemw** [comment]
Flexible input shape feature has been added to CoreML. 
tfcoreml needs a fixed shape while conversion and will produce a CoreML model with fixed shape. 
However, using coremltools (>= 2.0b1), you can easily add shape flexibility to a .mlmodel.

**Comment by dimroc** [comment]
Thanks for the info @aseemw 

> However, using coremltools (>= 2.0b1), you can easily add shape flexibility to a .mlmodel.

Can you point us in the direction as to how to do this? Out of the box, my keras model with `Input(shape=(None, None, 3))` does not work with `coremltools==2.0b1`

**Comment by cbecker** [comment]
@aseemw can you point to the document that explains how to use flexible input shape in CoreML? I am dealing with this issue as well, although it's not on TF.

**Comment by sndre** [comment]
@cbecker, here's a helpful snippet to get you started:

```python
spec = coremltools.utils.load_spec('Model.mlmodel')
img_size_ranges = flexible_shape_utils.NeuralNetworkImageSizeRange()
img_size_ranges.add_height_range((64, -1))
img_size_ranges.add_width_range((64, -1))
flexible_shape_utils.update_image_size_range(spec, feature_name='%INPUT_TENSOR_NAME%, size_range=img_size_ranges)
coremltools.models.utils.save_spec(spec, 'Model.mlmodel')
```

**Comment by cbecker** [comment]
Thanks @sndre . It seems like this is part of MLModel format v3, which can be loaded with MacOS 10.14 which is in beta now. Do you know if there's another way to allow for dynamic size models with 10.13?

**Comment by sndre** [comment]
@cbecker, I believe it's only supported on macOS 10.14 and iOS 12, but I would double check with @aseemw

**Comment by aseemw** [comment]
Yes that is correct, to use the new functionality of shape flexibility you would need the new operating systems as stated by sndre.

**Comment by vonchenplus** [comment]
@aseemw Is this just resize input?
